language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/OptimizeImportsCodeAction.java

```diff
@@ -106,8 +109,26 @@ public OptimizeImportsCodeAction() {
         }
 
         // Re-create imports list text
-        StringJoiner editText = new StringJoiner(System.lineSeparator());
-        sortImports(fileImports).forEach(importNode -> editText.add(importNode.toSourceCode()));
+        StringJoiner editText = new StringJoiner("");
+        sortImports(fileImports).forEach(importNode -> {
+            MinutiaeList leadingMinutiae = NodeFactory.createEmptyMinutiaeList();
+            MinutiaeList trailingMinutiae = importNode.importKeyword().trailingMinutiae();
+            Token modifiedImportKeyword = importNode.importKeyword().modify(leadingMinutiae, trailingMinutiae);
+
+            ImportDeclarationNode.ImportDeclarationNodeModifier importModifier = importNode.modify();
+            importModifier.withImportKeyword(modifiedImportKeyword);
+            if (importNode.orgName().isPresent()) {
+                importModifier.withOrgName(importNode.orgName().get());
+            }
+            importModifier.withModuleName(importNode.moduleName());
+            if (importNode.prefix().isPresent()) {
+                importModifier.withPrefix(importNode.prefix().get());
+            }
+            importNode.semicolon();
+            importModifier.withSemicolon(importNode.semicolon());
+
+            editText.add(importModifier.apply().toSourceCode());
```

Comment:
I feel like we have overly complicated things just to use `.toSourceCode()`. Could have done with few lines with this util class `CodeActionModuleId.from(importNode)`. We just need a toString() or can invoke getters. Shall we look that @nadeeshaan wdyt?

Reply:
`.toString()` also includes minutiae. So with that approach, I think we have to use importToken.text() for the import keyword and .toString() for the rest of the elements in the import declaration and make this even more complicated?

Reply:
@lochana-chathura I meant CodeActionModuleId.toString()

Reply:
I was following this approach since we directly manipulate the Syntax API and later we can use the same implementation to handle the comments in between in a proper manner.

Reply:
But when we modifying with `createEmptyMinutiaeList` for the leading minutiae, all comments are getting removed right?

Reply:
And how do we differentiate license header from the rest of comments in the minutiae?

Reply:
I'm just wondering if this code action supposed to do formatting also.
For example. if the input is 
```bal
import ballerina/xyz;
<new line>
import projectls.lsmod2;
```
Do we expect to preserve the newline in between, while changing the order? 
Leading minutiae includes new lines as wells. In above example middle new line will be leading minutiae of the second import keyword.

Reply:
@lochana-chathura code-action is supposed to preserve new lines. @nadeeshaan shall we move ahead with `.toSourceCode()` and just replace the leading minutiae of the **first** import node(before sorting). leading minutiae in other import nodes will preserve new lines.